### PR TITLE
Add env variables to control modules of api-server

### PR DESCRIPTION
**Pull request by wejdross**
## Summary

* Adding logic to enable specific features of appcat-apiserver based on env variables

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,23 +17,48 @@ limitations under the License.
...
-	err := builder.APIServer.
-		// +kubebuilder:scaffold:resource-register
-		WithResourceAndHandler(&appcatv1.AppCat{}, appcat.New()).
-		WithResourceAndHandler(&appcatv1.VSHNPostgresBackup{}, postgres.New()).
-		WithoutEtcd().
+
+	var appcatEnabled bool
+	var vshnEnabled bool
+
+	if os.Getenv("APPCAT_ENABLED") == "1" {
```
Comment (zugao) [inline main.go:37]:
```suggestion
	if os.Getenv("APPCAT_HANDLER_ENABLED") == "1" {
```

I would change a bit the name to be more explicit.

File: main.go [hunk 2]
```diff
@@ -17,23 +17,48 @@ limitations under the License.
...
+		appcatEnabled = true
+	}
+	if os.Getenv("VSHN_ENABLED") == "1" {
```
Comment (zugao) [inline main.go:40]:
```suggestion
	if os.Getenv("VSHN_POSTGRES_BACKUP_HANDLER_ENABLED") == "1" {
```

I would not create an aggregate group just now. Let's go with more refined enabling system.

File: config/aggregated-apiserver.yaml [hunk 3]
```diff
@@ -33,6 +33,11 @@ spec:
             - "--feature-gates=APIPriorityAndFairness=false"
             - "--audit-log-maxage=0"
             - "--audit-log-maxbackup=0"
+          env:
```
Comment (zugao) [inline config/aggregated-apiserver.yaml:36]: I would add all env variables in a config map.

**Review by zugao — COMMENTED** [review]
It would be good to have these parameters explained via `appcat-apiserver --help`. Also they have to be explained in docs as well.

File: main.go [hunk 4]
```diff
@@ -25,15 +28,41 @@ import (
...
-	err := builder.APIServer.
-		// +kubebuilder:scaffold:resource-register
-		WithResourceAndHandler(&appcatv1.AppCat{}, appcat.New()).
-		WithResourceAndHandler(&appcatv1.VSHNPostgresBackup{}, postgres.New()).
-		WithoutEtcd().
+
+	var appcatEnabled bool = false
+	var vshnEnabled bool = false
+
+	if os.Getenv("APPCAT_HANDLER_ENABLED") == "1" {
```
Comment (glrf) [inline main.go:35]: Nitpick: I'd default to `true` for enabling the appcat API handler

File: main.go [hunk 5]
```diff
@@ -25,15 +28,41 @@ import (
...
+		appcatEnabled = true
+	}
+	if os.Getenv("VSHN_POSTGRES_BACKUP_HANDLER_ENABLED") == "1" {
+		vshnEnabled = true
+	}
```
Comment (glrf) [inline main.go:40]:
```suggestion
    vshnEnabled, err = strconv.ParseBool(os.Getenv("VSHN_POSTGRES_BACKUP_HANDLER_ENABLED"))
	if err != nil {
	    klog.Printf("Failed to parse VSHN_POSTGRES_BACKUP_HANDLER_ENABLED: %s", err.Error())
		vshnEnabled = false
	}
```

I'd suggest using `ParseBool` to set the environment variable. That way `true` or `True` or `TRUE` or `T` all still work.

File: main.go [hunk 6]
```diff
@@ -25,15 +28,41 @@ import (
...
+
+	builder.WithoutEtcd().
 		ExposeLoopbackAuthorizer().
-		ExposeLoopbackMasterClientConfig().
-		Execute()
+		ExposeLoopbackMasterClientConfig()
+
+	cmd, err := builder.Build()
 	if err != nil {
+		log.Fatal(err)
```
Comment (glrf) [inline main.go:62]: Please use `klog` if we already have it

**Review by glrf — COMMENTED** [review]
Could you also add a few words about the environment variables in the README or so? Otherwise the only way to figure out what envVars you can set is by reading the code

File: main.go [hunk 7]
```diff
@@ -25,15 +29,48 @@ import (
...
-		// +kubebuilder:scaffold:resource-register
-		WithResourceAndHandler(&appcatv1.AppCat{}, appcat.New()).
-		WithResourceAndHandler(&appcatv1.VSHNPostgresBackup{}, postgres.New()).
-		WithoutEtcd().
+
+	var appcatEnabled bool = true
+	var vshnEnabled bool = false
+	var err error
+
+	if os.Getenv("APPCAT_HANDLER_ENABLED") != "" {
```
Comment (glrf) [inline main.go:37]: You don't need this check anymore. If it's `""` `PraseBool` will just return an error.

File: main.go [hunk 8]
```diff
@@ -25,15 +28,44 @@ import (
...
+
+	var appcatEnabled bool = true
+	var vshnEnabled bool = false
+	var err error
+
+	appcatEnabled, err = strconv.ParseBool(os.Getenv("APPCAT_HANDLER_ENABLED"))
+	if err != nil {
+		klog.Fatal("Can't parse APPCAT_HANDLER_ENABLED env variable")
+	}
+	vshnEnabled, err = strconv.ParseBool(os.Getenv("VSHN_POSTGRES_BACKUP_HANDLER_ENABLED"))
```
Comment (Kidswiss) [inline main.go:40]: Nitpick: the variable name could be a bit confusing, as it doesn't contain `backup` and enables the backuphandler.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just the usual: the summary in the PR is empty :)

**Review by glrf — APPROVED** [review]

**Review by zugao — APPROVED** [review]
Haven't checked on a real cluster but the code looks good to me.

**Pull request merged**